tools/shared/src/main/scala/org/scalajs/core/tools/corelib/CoreJSLibs.scala

```diff
@@ -99,7 +113,27 @@ object CoreJSLibs {
       else "" // blank line preserves line numbers in source maps
     }
 
-    lines.mkString("", "\n", "\n")
+    val content = lines.mkString("", "\n", "\n")
+
+    outputMode match {
+      case OutputMode.ECMAScript51Global =>
+        content
+
+      case OutputMode.ECMAScript51Isolated =>
+        content
+          .replaceAll("ScalaJS\\.d\\.", "\\$d_")
```

Comment:
This code only applies to the contents of [scalajsenv.js](https://github.com/scala-js/scala-js/blob/master/tools/scalajsenv.js). We would of course never prevent you from writing the Scala code that you want. This is completely internal.